src/graphics/Screen.cpp

```diff
@@ -1609,7 +1745,8 @@ void Screen::handleOnPress()
     // If screen was off, just wake it, otherwise advance to next frame
     // If we are in a transition, the press must have bounced, drop it.
     if (ui->getUiState()->frameState == FIXED) {
-        ui->nextFrame();
+        // TODO: Figure out if the rest of this is needed for waking up
+        // ui->nextFrame();
```

Comment:
Does this need to be disabled? It prevents devices with only a user-button from cycling through the frames.

Reply:
A solution could be enabling user button cycling for devices with only one button. It was disabled because the user button press now has a new function. What do you think?

Reply:
From a quick look, it seems like next frame has been disabled in `handleOnPress()` for all devices, and is now intercepted in `handleInputEvent()`, as `meshtastic_ModuleConfig_CannedMessageConfig_InputEventChar_SELECT`, right?
Only a select few devices will have the correct inputs to use this new behavior. Both `handleInputEvent()` and `handleOnPress()` will need to be tied to some sort of config option. 

A macro maybe wouldn't a great choice here, because a lot of DIY devices (e.g. Heltec V3) may have a single user button, or may have a more complex keyboard, depending on how the device is built. It wouldn't be fair to inconvenience all these users by forcing them to custom-build firmware from now on.

A lot of this functionality might need to tied to a new user-setting? Possibly there is an existing user-setting which can be used in this particular case (determining how to handle input)?

If you haven't considered it already, I'd also think about how this code interacts with the existing canned-messages module. Also might be an idea to think about how users with canned-message inputs different than the T-Deck would interact with this new feature.

Reply:
> From a quick look, it seems like next frame has been disabled in `handleOnPress()` for all devices, and is now intercepted in `handleInputEvent()`, as `meshtastic_ModuleConfig_CannedMessageConfig_InputEventChar_SELECT`, right?

Yes.

> For the record, pressing the user-button right now doesn't toggle the scrolling either on most devices , as the press event is not generally connected to the canned message select event (I think?)

Okay. Linking those scroll functions to the canned message select input should be changed.

> A macro maybe wouldn't a great choice here, because a lot of DIY devices (e.g. Heltec V3) may have a single user button, or may have a more complex keyboard, depending on how the device is built. It wouldn't be fair to inconvenience all these users by forcing them to custom-build firmware from now on.
>
> A lot of this functionality might need to tied to a new user-setting? Possibly there is an existing user-setting which can be used in this particular case (determining how to handle input)?
>
> If you haven't considered it already, I'd also think about how this code interacts with the existing canned-messages module. Also might be an idea to think about how users with canned-message inputs different than the T-Deck would interact with this new feature.

What do you mean by macro?

Reply:
I didn't realize macro refers to a C++ concept. I looked it up and figured it out. I have never coded in C++ before this so I'm learning as I go.

Reply:
I wouldn't worry too much about all these finer details right now to be honest. As long it's set up to only build for the T-Deck at first, all this stuff can wait.

Reply:
Alright, sounds good.